### PR TITLE
Added Recommendations Section

### LINKED ISSUE #10: Add Recommendations Section
Add a Testimonial (on LinkedIn, called Recommendations) Section.

**Layout**: Could either be an additional section or as part of the overlay/modal (#4) when you click on a project.

**Extend**: Once working solo, could experiment with obtaining profile info using the [LinkedIn API](https://www.linkedin.com/developers) to see if that retrieves recommendations to auto-populate.

### PR DESCRIPTION
**Pull request by louisefindlay23**
### Why This PR Adds Value

Added recommendations/testimonials section to showcase recommendations/testimonials from people you've worked for or with to provide social proof of your experience and skills.

### What This PR Adds

- This PR adds a new section to the main page called recommendations with two new files - recommendations.html and recommendations.yml.

- New styles are added to all.scss using the new .card .full class to hold styling for a full-width card.

- Added new config variable linkedin_username to provide a link to a person's LinkedIn profile for use in the testimonial button

Portfolio Recommendation Section

### How This PR Could Be Improved

- **Responsiveness** - CSS Media queries need to be added to improve the responsiveness of the full-width card. Possibly through a one-column card stacking layout. Related to #8 so wish to wait (or merge and handle this through that issue) until that's been completed in order to not conflict though two different approaches.

- **LinkedIn URL** - It would be nice if the URL for the LinkedIn button took users straight to the Recommendation section (and then the link could be moved to the entire card instead of a button) but research and attempts indicate that LinkedIn have updated their CSS IDs so they are automatically generated on each page load so there's no unique ID for an anchor link

- **Line Breaks** - Currently using `<br>` tags for line breaks in YAML. Would like to amend this to a more semantically better way if anyone knows how to do this in YAML.

- **LinkedIn API** - Use the LinkedIn API to grab a user's recommendations automatically. Requires an API key for user profile so will need to wait until we fork our own to do it individually.

### Issue This PR Closes

This closes #10.

### CODE REVIEW

File: _includes/recommendations.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <div class="card full">
+      <div class="icon">
+        <img src="./assets/img/recommendations/{{ item.logo }}">
+      </div>
+      <div class="info">
+        <p class="title">{{ item.title }}</p>
+        <p class="location">{{ item.author }}</p>
+        <p class="text">{{ item.text }}</p>
+        <button><a href="https://www.linkedin.com/in/{{ site.linkedin_username }}" target="_blank" rel="noopener">View more on LinkedIn</a></button>
+      </div>
```
Comment (jagnani73) [inline _includes/recommendations.html]: I think the `rel` should be `noopener noreferrer`
Can @robzwolf confirm

Comment (louisefindlay23) [reply _includes/recommendations.html]: Thanks, @jagnani73 for noticing that. Yes, it should be for wider browser support. Will amend.

Comment (louisefindlay23) [reply _includes/recommendations.html]: Have committed the changes. Thanks again.

**Comment by louisefindlay23** [comment]
Whoops, haven't branched off when committing the PR template update. Will remove the file and rebranch to make a new PR for it.

**Comment by Primata** [comment]
On LinkedIn authentication. Would each person have to create a personal[ application](https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow?context=linkedin%2Fcontext&tabs=HTTPS) for API requests and obtain Client ID and Client Secret? How would the oauth process be? I can't figure out how a static website would be granted authorization to display info.

**Comment by robzwolf** [comment]
@louisefindlay23 Is this ready for review?

**Comment by louisefindlay23** [comment]
> On LinkedIn authentication. Would each person have to create a personal[ application](https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow?context=linkedin%2Fcontext&tabs=HTTPS) for API requests and obtain Client ID and Client Secret? How would the oauth process be? I can't figure out how a static website would be granted authorization to display info.

I'm not sure, @SnowPrimate what you're referring to. Hope this clarifies. My thoughts were since the portfolio is for one specific person, I figured we could hardcode the Client ID and Secret in a `.dotenv` file. Then if people fork the repo, they can add their own Client ID and Secrets.

Wasn't going to implement here (on my own fork - once we do it personally later this week) since I don't have a client ID and Secret for Will's account unless he gives us it.

**Comment by louisefindlay23** [comment]
> 
> 
> @louisefindlay23 Is this ready for review?

Yes, thanks, @robzwolf. Isn't responsive yet but thought that would be covered in #8 since it's just another card class and associated styles. Can work on it tomorrow though if that's a blocker to being merged.

**Comment by robzwolf** [comment]
I'll review this once you fix the merge conflicts that were introduced by merging other PRs 🙃 Just ping me when you're ready for review. 

Happy for responsiveness to be tackled in a separate Issue+PR.

**Comment by louisefindlay23** [comment]
Thanks, @robzwolf. The conflict should be resolved now. Was including the recommendation and contact HTML files. Including both changes resolved the conflict.

File: assets/img/recommendations/tom-goodman.webp:Zone.Identifier [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
```
Comment (robzwolf) [inline assets/img/recommendations/tom-goodman.webp:Zone.Identifier]: What's this file? Something leftover from WSL?

**Review by robzwolf — CHANGES_REQUESTED** [review]

File: assets/img/recommendations/tom-goodman.webp:Zone.Identifier (hunk 2 above)
Comment (louisefindlay23) [reply assets/img/recommendations/tom-goodman.webp:Zone.Identifier]: Ah, a weird file from a .webp file I added. Yeah, is something to do with WSL. Will delete.

Comment (louisefindlay23) [reply assets/img/recommendations/tom-goodman.webp:Zone.Identifier]: Should be deleted now, @robzwolf.

**Comment by robzwolf** [comment]
Looks good! Get the merge conflict fixed and I'll approve this PR.

**Comment by louisefindlay23** [comment]
Thanks, @robzwolf. Conflicts are too much for the web editor :( so I'll have to check it out using a Git app. Having issues with Working Copy so may have to wait until tomorrow and I can use Fork instead.

**Comment by louisefindlay23** [comment]
That should be it fixed, @robzwolf or not since deploy failed. Will check on recommendations.html tomorrow.

**Comment by louisefindlay23** [comment]
Ah. All the files seem to have gone missing from my PR. Will fix.

**Comment by louisefindlay23** [comment]
That should finally be it properly working now, @robzwolf. Have also reordered the CSS and added comments for structure (readability) to make things clearer.

**Review by robzwolf — APPROVED** [review]

**Pull request merged**